npm/networkPolicyController.go

```diff
@@ -61,6 +71,35 @@ func NewNetworkPolicyController(npInformer networkinginformers.NetworkPolicyInfo
 	return netPolController
 }
 
+// initializeDataPlane do all initialization tasks for data plane
+func (c *networkPolicyController) initializeDataPlane() error {
+	klog.Infof("Initiailize data plane. Clean up Azure-NPM chians and start reconcile iptables")
+
+	// TODO(jungukcho): will clean-up error handling codes to initialize iptables and ipset in a separate PR
+	// It is important to keep order to clean-up iptables and ipset.
+	// IPtables should be cleaned first to avoid failures to clean-up iptables due to "ipset is using in kernel" error
+	// 1. clean-up NPM-related iptables information and then running periodic processes to keep iptables correct
+	c.iptMgr.UninitNpmChains()
```

Comment:
Good suggestion, but I would like to use **utilruntime.HandleError** function for log instead of **SendErrorLogAndMetric**.
```go
        if err := c.iptMgr.UninitNpmChains(); err != nil {
		utilruntime.HandleError(fmt.Errorf("Failed to UninitNpmChains with err: %s", err))
	}

	// 2. then clean-up all NPM ipsets states
	if err := c.ipsMgr.DestroyNpmIpsets(); err != nil {
		utilruntime.HandleError(fmt.Errorf("Failed to DestroyNpmIpsets with err: %s", err))
	}
```

I think using "utilruntime.HandleError" is fine since the error message is sent in "UninitNpmChains()" function already. 
Further, I do not want to use "metric.SendErrorLogAndMetric" function more in NPM code except for really critical error,  especially when the message is duplicated since it is an expensive operation (it uses network to send messages).

We need to re-organize error messages and error handling parts in concise and simple way.
Let me know if you have other opinion.

Reply:
The **UninitNpmChains** assumes there are already AZURE-NPM chains and if it does not exist, it returns error.
```go
   if err := c.iptMgr.UninitNpmChains(); err != nil {
		utilruntime.HandleError(fmt.Errorf("Failed to UninitNpmChains with err: %s", err))
	}

	// 2. then clean-up all NPM ipsets states
	if err := c.ipsMgr.DestroyNpmIpsets(); err != nil {
		utilruntime.HandleError(fmt.Errorf("Failed to DestroyNpmIpsets with err: %s", err))
	}
```

I want to keep this way in this PR and resolve this issue in a next PR since there are multiple places where two functions and used and it needs to walk through them carefully. In addition, considering the size of this PR, I do not want to increase review burden in this PR, which will reduce productivity.
Let me know if you have another opinion.

Reply:
Okay, for time being this is good, I think it is okay to ignore the error in this stage, i do not see a error case which can be fatal in NPMs operation later on.